toolbox/fdc3-conformance/FDC3-1.2-Conformance-Test-Cases.md

```diff
@@ -0,0 +1,183 @@
+| B   | joinChannel        | `fdc3.getSystemChannels()`<br>Check channels are returned.<br>Call `fdc3.joinChannel()` on first non-global channel                                                                                                                                                                                                                               |
+| B   | Broadcast          | `fdc3.broadcast()` the instrument context.<br>`fdc3.broadcast()` a contact context.                                                                                                                                                                                                                                                               |
+| A   | Receive Context    | Instrument object matches the one broadcast in 2 above.<br>Contact object matches the one broadcast in 2 above.                                                                                                                                                                                                                                   |
+
+-  `UCFilteredContext2`: Perform above test
+-  `UCFilteredContext3`: Perform above test, except joining a _different_ channel. Check that you _don't_ receive anything.
+-  `UCUnsubscribe`: Perform above test, except that after joining, **A** then `unsubscribe()`s the context listener. Check that **A** _doesn't_ receive anything.
+-  `UCFilteredContext4`: Perform above test, except that after joining, **A** changes channel with a further _different_ channel.  Check that **A** _doesn't_ receive anything.
+
+## 3. App Channels 
+
+### App Channels Broadcast (Basic)
+
+| App | Step               | Details                                                                                                                                                        |
+|-----|--------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| A   | createChannel        |`fdc3.getOrCreateChannel("test-channel")`       |
+| A   | addContextListener |Call `testChannel.addContextListener(null, handler)`<br>Check listener object returned<br>Check that there is an `unsubscribe` function on the returned object  |
+| B   | createChannel        | `fdc3. getOrCreateChannel("test-channel")`   |
+| B   | Broadcast          | `testChannel.broadcast(<some instrument>)`   |
+| A   | Receive Context    | Instrument object matches the one broadcast in 2 above.      |
+
+-  `ACBasicUsage1` Perform above test 
+-  `ACBasicUsage2` Perform above test, but join channel first and then `testChannel.addContextListener()`
+-  `ACBasicUsage3` Do the app B steps first but in reverse order to populate the channel with context, check that A will receive the context after creating
```

Comment:
```suggestion
| A   | createChannel        |`const testChannel = await fdc3.getOrCreateChannel("test-channel")`       |
| A   | addContextListener |Call `testChannel.addContextListener(null, handler)`<br>Check listener object returned<br>Check that there is an `unsubscribe` function on the returned object  |
| B   | createChannel        | `const testChannel = fdc3. getOrCreateChannel("test-channel")`   |
| B   | Broadcast          | `testChannel.broadcast(<some instrument>)`   |
| A   | Receive Context    | Instrument object matches the one broadcast in 2 above.      |

-  `ACBasicUsage1` Perform above test 

| App | Step               | Details                                                                                                                                                        |
|-----|--------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|
| B   | createChannel        | `fdc3. getOrCreateChannel("test-channel")`   |
| B   | Broadcast          | `testChannel.broadcast(<some instrument>)`   |
| A   | createChannel        |`const testChannel = await fdc3.getOrCreateChannel("test-channel")`       |
| A   | getCurrentContext |Call `testChannel.getCurrentContext()`<br>Check that the Context object returned is of the expected type.  |

-  `ACBasicUsage2` Perform above test.
-  `ACBasicUsage3` Perform above test, but in the final step call `getCurrentContext with a type name. I.e.  `testChannel.getCurrentContext("fdc3.instrument")`.
```

Reply:
This is actually whats implemented for ACBasicUsage2 and ACBasicUsage3 - but I think they just duplicate ACContextHistoryTyped etc..

Hence, they should probably be removed (from both here and the implementation) after checking that they really are duplicates. @robmoffat

Reply:
@robmoffat can confirm `ACContextHistoryTyped` is a duplicate of `ACBasicUsage3`. The only difference is that ACBasicUsage3 is checking context is received for a single fdc3.instrument type, whereas ACContextHistoryTyped checks for two context types. 

Let me know which one you want to remove and I'll remove it from the code